components/chainhook-sdk/src/chainhooks/types.rs

```diff
@@ -776,9 +776,17 @@ pub enum StacksTrait {
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub struct StacksPrintEventBasedPredicate {
-    pub contract_identifier: String,
-    pub contains: String,
+#[serde(untagged)]
+pub enum StacksPrintEventBasedPredicate {
```

Comment:
I _**really**_ don't like that we have to duplicate the definition of `contract_identifier` here. If anyone knows a way around this, that would be sick. That's what lead me down the path of the breaking change PR (#379), but I guess it's probably worth it to not break the API so soon after releasing v1.

Reply:
Agreed. After toying with serde, it does not look like something supported. If we really wanted to avoid the repetition, we could move to something like this:
```rust
struct Predicate {
   contract_identifier: T,
   contains: Option<U>,
   matches: Option<V>
}
```
and associate a method to that struct to make it more usable.

Reply:
Yeah that is an option.

In this case I guess we could treat this as an "OR", so we match any that contain _or_ regex match if both values are provided. I think it maybe sets a weird precedence for how we do this in the future - it would feel clunky to have a predicate type with many Options that are only really intended to be used alone, but the user can specify all of them. And if we ever need to make them mutually exclusive (you can have `fieldA` OR `fieldB` specified in the predicate, but not both), we wouldn't be able to represent this in an openapi spec.

All of that makes me lean toward the current solution, but no particularly strongly. Do you have a preference, @lgalabru?

Reply:
> All of that makes me lean toward the current solution

Agreed!

Reply:
> All of that makes me lean toward the current solution

Agreed!